File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/discount/FixedAmountDiscount.java
package com.fixture.shop.discount;

import com.fixture.shop.models.Order;

/**
 * Subtracts a fixed amount from the total (floored at zero).
 */
public class FixedAmountDiscount implements DiscountStrategy {

    private final double amount;

    public FixedAmountDiscount(double amount) {
        this.amount = amount;
    }

    @Override
    public double apply(Order order) {
        return Math.max(0, order.rawTotal() - amount);
    }

    public double getAmount() {
        return amount;
    }
}

File: skill_sdk/tests/fixtures/ts-project/src/shop/discount/FixedAmountDiscount.ts
import { DiscountStrategy } from './DiscountStrategy';
import { Order } from '../models/Order';

/**
 * Subtracts a fixed amount from the total (floored at zero).
 */
export class FixedAmountDiscount implements DiscountStrategy {
  constructor(private readonly amount: number) {}

  apply(order: Order): number {
    return Math.max(0, order.rawTotal() - this.amount);
  }

  getAmount(): number {
    return this.amount;
  }
}
